cicecore/cicedynB/analysis/ice_history_shared.F90

```diff
@@ -672,64 +672,67 @@ subroutine construct_filename(ncfile,suffix,ns)
         iday = mday
         isec = msec - dt
 
-        if (write_ic) isec = msec
         ! construct filename
         if (write_ic) then
+           isec = msec
            write(ncfile,'(a,a,i4.4,a,i2.2,a,i2.2,a,i5.5,a,a)')  &
               incond_file(1:lenstr(incond_file)),'.',iyear,'-', &
-              imonth,'-',iday,'-',isec,'.',suffix
+              imonth,'-',iday,'-',isec,'.',trim(suffix)
         else
 
-         if (hist_avg .and. histfreq(ns) /= '1') then
-          if (histfreq(ns) == 'h'.or.histfreq(ns) == 'H') then
-           ! do nothing
-          elseif (new_year) then
-           iyear = iyear - 1
-           imonth = 12
-           iday = daymo(imonth)
-          elseif (new_month) then
-           imonth = mmonth - 1
-           iday = daymo(imonth)
-          elseif (new_day) then
-           iday = iday - 1
-          endif
-         endif
-
-         cstream = ''
+           if (hist_avg) then
+              if (histfreq(ns) == '1' .or. histfreq(ns) == 'h'.or.histfreq(ns) == 'H') then
+                 ! do nothing
+              elseif (new_year) then
+                 iyear = iyear - 1
+                 imonth = 12
+                 iday = daymo(imonth)
+              elseif (new_month) then
+                 imonth = mmonth - 1
+                 iday = daymo(imonth)
+              elseif (new_day) then
+                 iday = iday - 1
+              endif
+           endif
```

Comment:
When I looked at this code in the context of #551, I did not readily understand why we have to do those substractions here, and why we do not have to do any if histfreq = 1 or h...
Is it because the file names for averaged data are dated at the start of the averaging period ? I think a code comment explaining that would not hurt :)

Reply:
This is all part of a rather complex interaction between dates and history frequencies with a certain amount of assumption built-in.  Basically, if you are doing annual, monthly, or daily averages, the history will be written on the "new_flag".  At that point, you're in the next year, month, or day.  So you subtract "1" so when you construct the filename, the name is associated with the year, month, or day being averaged.  Of course, all this comes unglued if the histfreq_n is not 1 in terms of the filename.  The instantaneous stuff and histfreq='1' just writes out a file with the current date in it because there isn't a general way to create a meaningful filename.  Most of what I did tried to preserve what's there and expand how it works.  There are many things that could be further improved.